samples/samples/benchmarking.py

```diff
@@ -0,0 +1,128 @@
+import threading
+from google.cloud import spanner_v1
+import time
+import math
+
+from google.cloud.spanner_v1 import transaction, pool, _helpers
+from google.cloud import spanner
+import concurrent.futures
+
+# Define your Spanner instance and database information
+project_id = "your_project_id"
+instance_id = "your_instance_id"
+database_id = "your_database_id"
+spanner_client = spanner_v1.Client(project=project_id)
+
+# Create a Spanner database instance
+instance = spanner_client.instance(instance_id)
+pool = pool.FixedSizePool(size = 10, logging_enabled=True)
+database = instance.database(pool=pool, database_id=database_id, close_inactive_transactions=True)
+
+transaction_time = []
+
+
+def calculatePercentile(latencies):
+    # sort the latencies array
+    latencies.sort()
+
+    # calculate p50 (50th percentile)
+    p50Index = math.floor(0.5*len(latencies))
+    p50Latency = latencies[p50Index]
+
+    # calculate p90 (90th percentile)
+    p90Index = math.floor(0.9*len(latencies))
+    p90Latency = latencies[p90Index]
+
+    return [p50Latency, p90Latency]
+
+# [START spanner_query_data]
+def query_data(thread_id):
+    print("running thread ", thread_id)
+    start_time = time.time()
+    time.sleep(10)
+    with database.snapshot() as snapshot:
+        results = snapshot.execute_sql(
+            "SELECT 1 FROM Singers"
+        )
+
+    # for row in results:
+    #     print(row)
+
+    # for row in results:
+    #     print("SingerId: {}, FirstName: {}, LastName: {}".format(*row))
+
+    end_time = time.time()
+    transaction_time.append(end_time-start_time)
+# [END spanner_query_data]
+
+# [START spanner_batch_transaction]
+def batch_transaction(thread_id):
```

Comment:
Where is this function `batch_transaction` invoked? Similarly where is `insert_with_dml` invoked?